### ISSUE TITLE
Protections against high fee values

### ISSUE THREAD

**Issue by mDuo13** [XRPLF/xrpl-py#213]
A common mistake people make is fat-fingering the `Fee` value of a transaction, or letting a script auto-fill it during a brief moment of very high load. There have been several cases of people burning whole XRP (or even thousands of XRP!) to send payments for fractions of that value.

By default, ripple-lib won't let you sign any payment with a max fee higher than a default limit of 2 XRP. (You can configure it in the [API constructor parameters, `maxFeeXRP`](https://xrpl.org/rippleapi-reference.html#parameters).) Even 2 XRP is a pretty high fee given that most transactions only need about 0.000010 XRP and even a [special case](https://xrpl.org/transaction-cost.html#special-transaction-costs) like an EscrowFinish with a big fulfillment probably doesn't go above 0.001000 XRP.

The only big exception is AccountDelete, which requires a full 5 XRP. But deleting accounts is (and probably should be) a rare, unusual operation so it makes sense to require the user to use a special override to sign a transaction with a `Fee` value that high.

In short, **We should raise an error when constructing or signing a transaction whose fee value is more than 2 whole XRP** unless the user explicitly specifies an option to allow a higher maximum fee.

**Comment by florent-uzio** [comment]
I am working on this one @mvadari and @ledhed2222 fyi
